### PR TITLE
LOOP-26 Upload Loop dosing decision data to Tidepool backend

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-26
- Improve StoredDosingDecision storage
- Move BolusRecommendation and BolusRecommendationNotice from Loop into LoopKit
- Make public operators of BolusRecommendation and BolusRecommendationNotice

### CODE REVIEW

File: Loop/Managers/LoopDataManager.swift [hunk 1]
```diff
@@ -653,32 +653,35 @@ extension LoopDataManager {
...
-            if let (recommendation: recommendation, date: date) = state.recommendedTempBasal {
-                dosingDecision.tempBasalRecommendationDate = TempBasalRecommendationDate(recommendation: recommendation, date: date)
-            }
-            dosingDecision.recommendedBolus = state.recommendedBolus?.recommendation.amount
-            if let lastReservoirValue = manager.doseStore.lastReservoirValue {
-                dosingDecision.lastReservoirValue = LastReservoirValue(startDate: lastReservoirValue.startDate, unitVolume: lastReservoirValue.unitVolume)
-            }
-            dosingDecision.pumpManagerStatus = self.delegate?.pumpManagerStatus
-            dosingDecision.error = error ?? state.error
+    func storeDosingDecision(withDate date: Date = Date(), withError error: Error? = nil) {
```
Comment (rickpasetto) [inline Loop/Managers/LoopDataManager.swift:656]: Out of curiosity: is this under unit test already?

**Review by rickpasetto — APPROVED** [review]
Just wondering if this is under unit test already, so this refactor still passes?  If so, LGTM.

File: Loop/Managers/LoopDataManager.swift (hunk 1 above)
Comment (darinkrauss) [reply Loop/Managers/LoopDataManager.swift:656]: There are already tests for storing a dosing decision.

**Review by nhamming — APPROVED** [review]
LGTM

**Pull request merged**